pandas/core/internals/array_manager.py

```diff
@@ -0,0 +1,896 @@
+                        #  obj.axes[-1].equals(self.items)
+                        if obj.ndim == 1:
+                            kwargs[k] = obj.iloc[i]
+                        else:
+                            kwargs[k] = obj.iloc[:, i]._values
+                    else:
+                        # otherwise we have an array-like
+                        kwargs[k] = obj[i]
+
+            try:
+                if callable(f):
+                    applied = f(arr, **kwargs)
+                else:
+                    applied = getattr(arr, f)(**kwargs)
+            except (TypeError, NotImplementedError):
+                if not ignore_failures:
+                    raise
+                continue
+            # if not isinstance(applied, ExtensionArray):
+            #     # TODO not all EA operations return new EAs (eg astype)
+            #     applied = array(applied)
+            result_arrays.append(applied)
+            result_indices.append(i)
+
+        new_axes: List[Index]
+        if ignore_failures:
+            # TODO copy?
+            new_axes = [self._axes[0], self._axes[1][result_indices]]
+        else:
+            new_axes = self._axes
+
+        if len(result_arrays) == 0:
+            return self.make_empty(new_axes)
+
+        return type(self)(result_arrays, new_axes)
+
+    def apply_with_block(self: T, f, align_keys=None, **kwargs) -> T:
+
+        align_keys = align_keys or []
+        aligned_args = {k: kwargs[k] for k in align_keys}
+
+        result_arrays = []
+
+        for i, arr in enumerate(self.arrays):
+
+            if aligned_args:
+                for k, obj in aligned_args.items():
+                    if isinstance(obj, (ABCSeries, ABCDataFrame)):
+                        # The caller is responsible for ensuring that
+                        #  obj.axes[-1].equals(self.items)
+                        if obj.ndim == 1:
+                            kwargs[k] = obj.iloc[[i]]
+                        else:
+                            kwargs[k] = obj.iloc[:, [i]]._values
+                    else:
+                        # otherwise we have an ndarray
+                        kwargs[k] = obj[[i]]
+
+            if hasattr(arr, "tz") and arr.tz is None:  # type: ignore[union-attr]
+                # DatetimeArray needs to be converted to ndarray for DatetimeBlock
+                arr = arr._data  # type: ignore[union-attr]
+            elif arr.dtype.kind == "m":
+                # TimedeltaArray needs to be converted to ndarray for TimedeltaBlock
+                arr = arr._data  # type: ignore[union-attr]
+            if isinstance(arr, np.ndarray):
+                arr = np.atleast_2d(arr)
+            block = make_block(arr, placement=slice(0, 1, 1), ndim=2)
+            applied = getattr(block, f)(**kwargs)
+            while isinstance(applied, list):
+                # ObjectBlock gives double nested result?, some functions give no list
```

Comment:
this is weird.  can you give an example?

Reply:
> this is weird. can you give an example?

Might be something that is fixed in the meantime. Don't remember exactly, and when changing to a simple `if` instead of the `while`, the subset of tests that is ran on CI still seems to pass. So will simplify here.